src/raw_iter.rs

```diff
@@ -62,31 +62,70 @@ where
+/// Iterator over dynamic color. Positions are dependent on the row order.
+#[allow(missing_debug_implementations)]
+pub enum DynamicRawColors<'a> {
+    /// 1 bit per pixel
     Bpp1(RawColors<'a, RawU1>),
+    /// 4 bit per pixel
     Bpp4(RawColors<'a, RawU4>),
+    /// 8 bit per pixel
     Bpp8(RawColors<'a, RawU8>),
+    /// 16 bit per pixel
     Bpp16(RawColors<'a, RawU16>),
+    /// 24 bit per pixel
     Bpp24(RawColors<'a, RawU24>),
+    /// 32 bit per pixel
     Bpp32(RawColors<'a, RawU32>),
-    Bpp4Rle(Rle4Pixels<'a>),
-    Bpp8Rle(Rle8Pixels<'a>),
+    /// 4 bit per pixel RLE encoded
+    Bpp4Rle(Rle4Colors<'a>),
+    /// 8 bit per pixel RLE encoded
+    Bpp8Rle(Rle8Colors<'a>),
 }
 
-impl<'a> DynamicRawColors<'a> {
-    pub fn new(raw_bmp: &'a RawBmp<'a>) -> Self {
-        let header = raw_bmp.header();
-        match header.compression_method {
-            CompressionMethod::Rle4 => DynamicRawColors::Bpp4Rle(Rle4Pixels::new(raw_bmp)),
-            CompressionMethod::Rle8 => DynamicRawColors::Bpp8Rle(Rle8Pixels::new(raw_bmp)),
-            CompressionMethod::Rgb | CompressionMethod::Bitfields => match header.bpp {
-                Bpp::Bits1 => DynamicRawColors::Bpp1(RawColors::new(raw_bmp)),
-                Bpp::Bits4 => DynamicRawColors::Bpp4(RawColors::new(raw_bmp)),
-                Bpp::Bits8 => DynamicRawColors::Bpp8(RawColors::new(raw_bmp)),
-                Bpp::Bits16 => DynamicRawColors::Bpp16(RawColors::new(raw_bmp)),
-                Bpp::Bits24 => DynamicRawColors::Bpp24(RawColors::new(raw_bmp)),
-                Bpp::Bits32 => DynamicRawColors::Bpp32(RawColors::new(raw_bmp)),
-            },
+impl core::fmt::Debug for DynamicRawColors<'_> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            DynamicRawColors::Bpp1(_) => f.debug_tuple("DynamicRawColors::Bpp1").finish(),
+            DynamicRawColors::Bpp4(_) => f.debug_tuple("DynamicRawColors::Bpp4").finish(),
+            DynamicRawColors::Bpp8(_) => f.debug_tuple("DynamicRawColors::Bpp8").finish(),
+            DynamicRawColors::Bpp16(_) => f.debug_tuple("DynamicRawColors::Bpp16").finish(),
+            DynamicRawColors::Bpp24(_) => f.debug_tuple("DynamicRawColors::Bpp24").finish(),
+            DynamicRawColors::Bpp32(_) => f.debug_tuple("DynamicRawColors::Bpp32").finish(),
+            DynamicRawColors::Bpp4Rle(_) => f.debug_tuple("DynamicRawColors::Bpp4Rle").finish(),
+            DynamicRawColors::Bpp8Rle(_) => f.debug_tuple("DynamicRawColors::Bpp8Rle").finish(),
+        }
+    }
+}
+
+impl DynamicRawColors<'_> {
+    /// Get the row order of the bitmap.
+    pub fn row_order(&self) -> RowOrder {
+        match self {
+            DynamicRawColors::Bpp1(_)
+            | DynamicRawColors::Bpp4(_)
+            | DynamicRawColors::Bpp8(_)
+            | DynamicRawColors::Bpp16(_)
+            | DynamicRawColors::Bpp24(_)
+            | DynamicRawColors::Bpp32(_) => RowOrder::TopDown,
```

Comment:
Uncompressed BMP files can be top down or bottom up. It should be enough to return the value in `Header::row_order` for both, uncompressed and RLE files.